sources/org.osbuild.librepo

```diff
@@ -0,0 +1,242 @@
+#!/usr/bin/python3
+"""
+Source for downloading rpms using librepo.
+
+Download the list of rpms using a metalink or mirrorlist URL, trying new
+mirrors if there is an error. The files are written to the osbuild file cache
+using the hash as the filename.
+
+It can download files that require secrets. The only secret provider currently
+supported is `org.osbuild.rhsm` for downloading Red Hat content that requires a
+subscriptions.
+"""
+
+import sys
+from typing import Dict
+
+import librepo
+
+from osbuild import sources
+from osbuild.util.rhsm import Subscriptions
+
+# NOTE: The top level schema properties are limited to items and options by the
+# v2 schema definition
+SCHEMA_2 = """
+"properties": {
+  "items": {
+        "description": "List of the packages and their hash to download from the mirror",
+        "type": "object",
+        "additionalProperties": false,
+        "patternProperties": {
+          "(md5|sha1|sha256|sha384|sha512):[0-9a-f]{32,128}": {
+             "required": [
+               "path", "mirror"
+             ],
+             "properties": {
+               "path": {
+                 "description": "Name or path of the package file. Supports bare name or relative paths",
+                 "type": "string"
+               },
+               "mirror": {
+                 "description": "The mirror to use for this package",
+                 "type": "string"
+               }
```

Comment:
The name of this property tripped me up at first.  I was thinking "why specify a mirror when the point is to use multiple mirrors for each package?", until I realised this is actually referring to one of the _mirrorlists_ defined below.  Would it make more sense to rename this to `mirrorlist` and the property in the options below to `mirrorlists`?

Reply:
It's actually a mirror id, referencing the keys in `options.mirrors` so maybe `id` or `mirror_id` would be clearer? `mirrorlist` implies an actual mirror list, which is different from a metalink :) So I thing of `mirror` being a generic word covering all the types.

Reply:
Right, I see.  `mirror_id` might make it clearer, yes.  But I think given your explanation it's also okay to leave it as is.